suse_migration_services/units/prepare.py

```diff
@@ -251,3 +262,29 @@ def get_root_disk_device(root_path):
                 block_type = block_record[1]
                 if block_type in considered_block_types:
                     return block_record[0]
+
+
+def get_regionsrv_client_file_location(root_path):
+    """
+    Find the correct path for the cloud-regionsrv-client cache files
+    as the location moves from /var/lib/cloudregister
+    (pre cloud-regionsrv-client-10.0.4) to /var/cache/cloudregister
+    (post cloud-regionsrv-client-10.0.4)
+    """
+    old_cloud_register_path = os.sep.join(
+        [root_path, 'var', 'lib', 'cloudregister']
+    )
+    new_cloud_register_path = os.sep.join(
```

Comment:
you do not need two variables

maybe consider this approach ?

```python
    # old path
    cloud_register_path = os.sep.join(
        [root_path, 'var', 'lib', 'cloudregister']
    )
    if os.path.isdir(cloud_register_path) and \
            any('SMT' in x for x in os.listdir(cloud_register_path)):
        return cloud_register_path
    # new path
    cloud_register_path = os.sep.join(
        [root_path, 'var', 'cache', 'cloudregister']
    )
    if os.path.isdir(cloud_register_path) and \
            any('SMT' in x for x in os.listdir(cloud_register_path)):
        return cloud_register_path
```

Reply:
i fixed this, please check